modules/calc/src/main/java/com/opengamma/strata/calc/runner/AbstractDerivedCalculationFunction.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.calc.runner;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.strata.basics.CalculationTarget;
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.calc.Measure;
+
+/**
+ * Abstract derived calculation function with fields for the target type, measure and required measures.
+ * <p>
+ * Empty requirements are returned from {@link #requirements}.
+ * Subtypes only need to provide an implementation of the {@link #calculate} method.
+ */
+public abstract class AbstractDerivedCalculationFunction<T extends CalculationTarget, R>
```

Comment:
I thought it was useful. Otherwise every implementation will have to implement at least those three methods, and possibly the corresponding fields too. This way there is just the constructor call to the superclass to implement.